packages/yoroi-ergo-connector/example/index.js

```diff
@@ -104,15 +123,34 @@ function initDapp() {
                         wasm.Address.from_base58(changeAddr),
                         wasm.BoxValue.SAFE_USER_MIN());
                         //changeValueBoxValue);
-                    
+                    const dataInputs = new wasm.DataInputs();
+                    // ranndom tx we sent via the connector before - not referenced in any smart contract right now
+                    //dataInputs.add(new wasm.DataInput(wasm.BoxId.from_str("0f0e4c71ccfbe7e749591ef2a906607b415deadee8c23a8d822517c4cd55374e")));
+                    txBuilder.set_data_inputs(dataInputs);
                     const tx = txBuilder.build().to_json();
-                    console.log(`tx: ${tx}`);
+                    console.log(`tx: ${JSON.stringify(tx)}`);
+                    console.log(`original id: ${tx.id}`);
+                    // sigma-rust doesn't support most compilation so manually insert it here
+                    // this is HEIGHT > 1337 but in hex and without the checksum/etc for the address of the contract
+                    //tx.outputs[0].ergoTree = "100104f214d191a37300";
+                    // and we rebuild it using
+                    const correctTx = wasm.UnsignedTransaction.from_json(JSON.stringify(tx)).to_json();
```

Comment:
What is the reason for JSON roundtrip for UnsignedTransaction? Do `original id:` and `new id:` have different ids?

Reply:
It was for line 135 - I was adding a P2S address as an output to test something, and wasn't able to do it via sigma-rust directly so I just manually changed one of the ergoTrees then using my modified fork used just for the dapp for testing of sigma-rust that ignores the txid and recalculates it upon from_json I do the round-trip to recalculate the txid to be the proper one with that new ergotree. The resulting TX was accepted by the network, the problem was when we tried to spend it. I think I've found the problem for the different IDs though, I'll try to push a new commit soon. It was inside of Yoroi not the dApp.

Reply:
> Do `original id:` and `new id:` have different ids?

When UnsignedTransaction is parsed from JSON it recalculates the tx id and if it differs from the one provided in JSON it fails the parsing process. So I wonder how they could be different in the logs.

Reply:
They aren't different between the two logs here, it was different between here and the resulting signed tx spit out by Yoroi. What was happening was when the signing logic was written we needed some internal datastructure in Yoroi for unspent boxes to get the keys, and to do that we filtered the user's UTXOs based on the inputs we needed to sign, which since a P2S address would never be inside the user's tracked UTXOs would silently filter that out during signing. It's fixed now though! The TX is accepted by the network: https://explorer.ergoplatform.com/en/transactions/59b94cfc3244d1358a2b06a1abaab2ee16884e2589554c42a593cf8ecffbb33e